### PR TITLE
BootManager: Slightly clean up the config_cache.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
Very slightly, just really pulling the save and restore tasks into methods instead of having them in the middle of a different function. This indirectly fixes the same thing #2619 fixes, and is supposed to be preventive so a similar thing doesn't happen again.

I haven't actually checked if this error exists as I think it does, but this should also fix an error where settings from movie playbacks or netplay sessions playing homebrew would overwrite the global settings.

### CODE REVIEW

**Comment by lioncash** [comment]
It's not necessary to use `this->` especially when there's no name clashing.

File: Source/Core/Core/BootManager.cpp [hunk 1]
```diff
@@ -57,8 +58,136 @@ struct ConfigCache
...
+	std::fill_n(this->bSetWiimoteSource, (int)MAX_BBMOTES, false);
+	this->bSetFrameSkip = false;
+}
+
+void ConfigCache::RestoreConfig(SConfig& config)
+{
+	if (!valid)
+	{
+		return;
+	}
```
Comment (lioncash) [inline Source/Core/Core/BootManager.cpp]: These braces are not necessary

File: Source/Core/Core/BootManager.cpp [hunk 2]
```diff
@@ -57,8 +58,136 @@ struct ConfigCache
...
+		Movie::SetFrameSkipping(this->frameSkip);
+	}
+	for (unsigned int i = 0; i < MAX_EXI_CHANNELS; ++i)
+	{
+		if (this->bSetEXIDevice[i])
+		{
+			config.m_EXIDevice[i] = this->m_EXIDevice[i];
+		}
+	}
+
```
Comment (lioncash) [inline Source/Core/Core/BootManager.cpp]: There should be spacing between all of these control structures.

**Comment by degasus** [comment]
What do you think about operator= instead?

**Comment by lioncash** [comment]
Having a named function is much more immediately readable IMO (although I would return the changed instance, instead of modifying in place)

**Comment by AdmiralCurtiss** [comment]
SConfig is currently not copyable for whatever reason, so we'd need to remove that if we don't want to modify in place. But yeah that's probably a bit cleaner.

**Comment by degasus** [comment]
LGTM

File: Source/Core/Core/BootManager.cpp [hunk 3]
```diff
@@ -57,8 +58,138 @@ struct ConfigCache
...
+	// these store if the relevant setting should be reset back later (true) or if it should be left alone on restore (false)
 	bool bSetFramelimit, bSetEXIDevice[MAX_EXI_CHANNELS], bSetVolume, bSetPads[MAX_SI_CHANNELS], bSetWiimoteSource[MAX_BBMOTES], bSetFrameSkip;
+
+	// fill the cache with values from the configuration
+	void SaveConfig(SConfig& config);
+	// restore values to the configuration from the cache
+	void RestoreConfig(SConfig& config);
 };
+
+void ConfigCache::SaveConfig(SConfig& config)
```
Comment (Tilka) [inline Source/Core/Core/BootManager.cpp]: `const SConfig&` please.

File: Source/Core/Core/BootManager.cpp [hunk 4]
```diff
@@ -57,8 +58,138 @@ struct ConfigCache
...
+
+	bSetFramelimit = false;
+	std::fill_n(bSetEXIDevice, (int)MAX_EXI_CHANNELS, false);
+	bSetVolume = false;
+	std::fill_n(bSetPads, (int)MAX_SI_CHANNELS, false);
+	std::fill_n(bSetWiimoteSource, (int)MAX_BBMOTES, false);
+	bSetFrameSkip = false;
+}
+
+void ConfigCache::RestoreConfig(SConfig& config)
```
Comment (Tilka) [inline Source/Core/Core/BootManager.cpp]: `SConfig*` please.

File: Source/Core/Core/BootManager.cpp [hunk 5]
```diff
@@ -58,8 +59,138 @@ struct ConfigCache
 	TEXIDevices m_EXIDevice[MAX_EXI_CHANNELS];
 	std::string strBackend, sBackend;
 	std::string m_strGPUDeterminismMode;
+
+public:
```
Comment (JosJuice) [inline Source/Core/Core/BootManager.cpp]: public needs to go before private.

**Comment by JosJuice** [comment]
Other than that, LGTM. Fixes stuff.

File: Source/Core/Core/BootManager.cpp (hunk 5 above)
Comment (phire) [inline Source/Core/Core/BootManager.cpp]: Rules about the ordering of public and private are a little stupid.

**Comment by AdmiralCurtiss** [comment]
So should I reorder that or not?

**Comment by phire** [comment]
Your choice, I'll merge it either way.

File: Source/Core/Core/BootManager.cpp (hunk 5 above)
Comment (spxtr) [inline Source/Core/Core/BootManager.cpp]: They are, but IMO [this](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/InputCommon/ControllerInterface/evdev/evdev.h#L20-L83) is pretty terrible, even though it makes sense why they did it that way.

**Comment by AdmiralCurtiss** [comment]
Might as well.

File: Source/Core/Core/BootManager.cpp [hunk 6]
```diff
@@ -48,7 +48,17 @@ namespace BootManager
 // Apply fire liberally
 struct ConfigCache
 {
-	bool valid, bCPUThread, bSkipIdle, bSyncGPUOnSkipIdleHack, bFPRF, bAccurateNaNs, bMMU, bDCBZOFF, m_EnableJIT, bDSPThread,
+public:
+	// fill the cache with values from the configuration
+	void SaveConfig( const SConfig& config );
```
Comment (lioncash) [inline Source/Core/Core/BootManager.cpp]: These shouldn't have spaces by the parentheses

**Pull request merged**